cluster-autoscaler/core/scale_down.go

```diff
@@ -1031,25 +1040,36 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
 
 	for range pods {
 		select {
-		case err := <-confirmations:
-			if err != nil {
-				evictionErrs = append(evictionErrs, err)
+		case evictionResult := <-confirmations:
+			if evictionResult.err != nil {
+				evictionErrs = append(evictionErrs, evictionResult.err)
 			} else {
+				delete(podsNotYetEvicted, evictionResult.pod.Name)
 				metrics.RegisterEvictions(1)
 			}
 		case <-time.After(retryUntil.Sub(time.Now()) + 5*time.Second):
+			// All successfully evicted pods were deleted from podsNotYetEvicted, so after a timeout it contains pods that either failed to evict or haven't evicted in time.
+			failingPods := make([]*apiv1.Pod, 0, len(podsNotYetEvicted))
+			for _, pod := range podsNotYetEvicted {
+				failingPods = append(failingPods, pod)
+			}
 			return errors.NewAutoscalerError(
-				errors.ApiCallError, "Failed to drain node %s/%s: timeout when waiting for creating evictions", node.Namespace, node.Name)
+				errors.ApiCallError, "Failed to drain node %s/%s: timeout when waiting for creating evictions", node.Namespace, node.Name), status.NodeDeleteErrorFailedToEvictPods, failingPods
```

Comment:
Looks like we lose information about eviction errors collected so far in this case. It may be OK, just wondering if it's a conscious decision (other than "we never logged them before").

Reply:
Not anymore!